packages/grid/x-grid/src/DataGridPro.tsx

```diff
@@ -57,4 +57,12 @@ export const DataGridPro = React.memo(DataGridProRaw);
 DataGridProRaw.propTypes = {
   columns: PropTypes.array.isRequired,
   rows: PropTypes.array.isRequired,
+  checkboxSelectionVisibleOnly: chainPropTypes(PropTypes.bool, (props: any) => {
+    if (!props.pagination && props.checkboxSelectionVisibleOnly) {
+      return new Error(
+        'Material-UI: The `checkboxSelectionVisibleOnly` prop has no effect when the pagination is not enabled.',
```

Comment:
A detail. I would advocate that a great warning starts by explaining what's wrong, and then explain how to solve the issue. We did the first one, but not the second. For instance, see how [React handles it](https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/scripts/error-codes/codes.json#L209).